Fix reservation test cases for large disks

behlendorf opened the pull request:
Convert explicit `typeset -i` declarations to `typeset -l` in order
to prevent 32-bit overflow from occurs with disks >2G.

behlendorf commented:
@gmelikov increasing the default volume size to 4G causes these test cases to fail due to 32-bit arithmatic limitations of ksh.  This patch disables these test case until they can be updated.  @jwk404 is this an issue for OpenZFS too?  If so how are you handling it?

jwk404 commented:
At first glance, I'd say  this is a problem I introduced by using `typeset  -i` here. The shell isn't limited to 32 bit math if you use `typeset -l foo` or even just `typeset foo` by itself. We're not seeing this in OpenZFS because these tests (only these) use bash instead of ksh. When the tests first integrated, I'd switched to bash but was asked in review to keep using ksh. Somehow this set of tests must have been overlooked.

behlendorf commented:
@jwk404 spot on!  Thanks for the pointer to `typeset -l`.  I've updated the patch to make the naive fix here and converted the `typeset -i` instances which could overflow to `typeset -l`.  This resolved the issue. 

I didn't notice you were using bash for theses scripts either.  Do you intend to switch them back to ksh?  If so what's your preferred fix for this?

jwk404 commented:
I'll convert our tests to use ksh, and at the same time, pull in your fix and we'll be back in sync again.

tuxoko commented:
But why `typeset -l`?
```
$ help typeset
-l	to convert NAMEs to lower case on assignment
```
Why not just remove the typeset.

gmelikov commented:
@tuxoko i've read that `typeset` makes variables local too.

@behlendorf thanks, i'll rebase #5679 just after this PR will be merged.

jwk404 commented:
@tuxoko @gmelikov The -l suggestion was a misreading of the manpage on my part. The typeset should be kept though, to keep the variables local.

behlendorf commented:
Right, so it looks like adding `-l` worked for the wrong reasons.  I've update the patch to drop `-l` and fixed a few other places which `-l` was used incorrectly.
